glib-macros/tests/object_subclass_dynamic.rs

```diff
@@ -11,13 +11,21 @@ mod static_ {
         // impl for an object interface to register as a static type.
         #[derive(Clone, Copy)]
         #[repr(C)]
-        pub struct MyStaticInterface {
+        pub struct MyStaticInterfaceClass {
             parent: glib::gobject_ffi::GTypeInterface,
         }
 
+        unsafe impl InterfaceClassStruct for MyStaticInterfaceClass {
+            type Type = MyStaticInterface;
+        }
+
+        pub struct MyStaticInterface;
```

Comment:
A value of this never appears anywhere so this could also be an uninhabited type or not?

```suggestion
        pub enum MyStaticInterface {}
```

Reply:
I like it because it makes it more explicit that there are never values of this type